### ISSUE TITLE
`2.20.12` Execution error […] / is not a relative path

### ISSUE THREAD

**Issue by eneroth** [thheller/shadow-cljs#1069]
I'm seeing something odd while trying to compile an uberjar where `2.20.12` is a dep.

Our dev builds has shadow-cljs compiled into them. As of `2.20.12`, something odd seems to happen when compiling `closure-compiler-unshaded-v20221102.jar` with `tools.build` (via `io.github.seancorfield/build-clj`). I don't really understand the internals of uberjar compilation, so this might be some actual error from the Closure compiler jar itself. 

That said, I've tried the previous Closure compiler version:
```clojure 
{thheller/shadow-cljs                   {:mvn/version "2.20.12"
                                         :exclusions  [com.google.javascript/closure-compiler]}
 com.google.javascript/closure-compiler {:mvn/version "v20220719"}}
```

And this gives the same error, so shadow-cljs itself is not entirely ruled out of the equation. `2.20.10` works fine.

Any idea what's going on here?

The exception is thrown by `clojure.tools.build.tasks.uber/explode`:

### Exception
```
Execution error (IllegalArgumentException) at clojure.tools.build.tasks.uber/explode (uber.clj:169).
/ is not a relative path
```

### `explode` inputs
```Clojure
{:lib-file #object[java.io.File 0xa389007 "<redacted>/.m2/repository/com/google/javascript/closure-compiler-unshaded/v20221102/closure-compiler-unshaded-v20221102.jar"],
 :lib      com.google.javascript/closure-compiler-unshaded,
 :opts     {:out-dir  #object[java.io.File 0x150a6caf "/var/folders/hw/fgtrn83s3d53gpj4tykp66k00000gn/T/uber10249887550530612444"],
            :handlers {#"^data_readers.clj[c]?$"
                       #object[clojure.tools.build.tasks.uber$conflict_data_readers 0x713dc01a "clojure.tools.build.tasks.uber$conflict_data_readers@713dc01a"],
                       #"^META-INF/services/.*"
                       #object[clojure.tools.build.tasks.uber$conflict_append 0xd173636 "clojure.tools.build.tasks.uber$conflict_append@d173636"],
                       #"(?i)^(META-INF/)?(COPYRIGHT|NOTICE|LICENSE)(\.(txt|md))?$"
                       #object[clojure.tools.build.tasks.uber$conflict_append_dedupe 0x6a503a3e "clojure.tools.build.tasks.uber$conflict_append_dedupe@6a503a3e"],
                       :default
                       #object[clojure.tools.build.tasks.uber$built_ins$fn__3575 0x1489257e "clojure.tools.build.tasks.uber$built_ins$fn__3575@1489257e"],
                       #"^META-INF/org/apache/logging/log4j/core/config/plugins/Log4j2Plugins.dat$"
                       #object[org.corfield.log4j2_conflict_handler$log4j2_plugin_merger 0x55d71296 "org.corfield.log4j2_conflict_handler$log4j2_plugin_merger@55d71296"]},
            :exclude  (#"project.clj"
                       #"META-INF/.*\.(?:SF|RSA|DSA|MF)"
                       #"module-info\.class"
                       #"(.*/)?\.DS_Store"
                       #"(.*/)?\.keep"
                       #".*\.pom"
                       #"(?i)META-INF/(?:INDEX\.LIST|DEPENDENCIES)(?:\.txt)?")}}
```

### `explode` loop state at time of failure
```Clojure
{;; From `(.getNextJarEntry jis)`
 :jar-entry #object[java.util.jar.JarEntry 0x4fb5f99e "/"]
 ;; From `(.getName entry)`
 :path "/",
 ;; From `opts` above
 :out-dir #object[java.io.File 0x150a6caf "/var/folders/hw/fgtrn83s3d53gpj4tykp66k00000gn/T/uber10249887550530612444"]}
```

**Comment by thheller** [comment]
I don't really know what this is, I didn't change anything in how the shadow-cljs jar is created so I suspect its one of the deps I changed in that release.

However, all of this really doesn't matter and the fix is something I recently explained in Slack.

**shadow-cljs should absolutely and under no circumstance ever end up in an uberjar.** It is a development tool and has no place in a production app. The likely problem is that you have `shadow-cljs` added as a dependency in your default `:deps` in `deps.edn`. Do not do that if you intend to build uberjars. Put it into an alias that is not activated when building the uberjar.


So, do not do this:
```clojure
{:deps
 {your.other-deps {...}
  thheller/shadow-cljs {...}}}
```

Do this instead:
```clojure
{:deps
 {your.other-deps {...}}
 :aliases
 {:cljs
  {:extra-deps
    {thheller/shadow-cljs {...}}}}
```

And if you have `:deps true` in `shadow-cljs.edn`, you change that to `:deps {:aliases [:cljs]}` accordingly. Also move all your other CLJS related dependencies to this alias, as they have no use in the uberjar either.
